opentracing-spring-web-autoconfigure/src/main/java/io/opentracing/contrib/spring/web/autoconfig/RestTemplateTracingAutoConfiguration.java

```diff
@@ -0,0 +1,174 @@
+package io.opentracing.contrib.spring.web.autoconfig;
+
+import io.opentracing.Tracer;
+import io.opentracing.contrib.spring.web.client.RestTemplateSpanDecorator;
+import io.opentracing.contrib.spring.web.client.TracingAsyncRestTemplateInterceptor;
+import io.opentracing.contrib.spring.web.autoconfig.client.TracingRestTemplateCustomizer;
+import io.opentracing.contrib.spring.web.client.TracingRestTemplateInterceptor;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.springframework.boot.autoconfigure.AutoConfigureAfter;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.boot.web.client.RestTemplateBuilder;
+import org.springframework.boot.web.client.RestTemplateCustomizer;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.http.client.AsyncClientHttpRequestInterceptor;
+import org.springframework.http.client.ClientHttpRequestInterceptor;
+import org.springframework.http.client.support.InterceptingAsyncHttpAccessor;
+import org.springframework.http.client.support.InterceptingHttpAccessor;
+import org.springframework.web.client.AsyncRestTemplate;
+import org.springframework.web.client.RestTemplate;
+
+import javax.annotation.PostConstruct;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Instrumentation of {@link RestTemplate} and {@link AsyncRestTemplate}.
+ * <p>
+ * For Spring Boot it also instruments {@link RestTemplateBuilder} and all instances created with it.
+ *
+ * @author Pavol Loffay
+ */
+@Configuration
+@ConditionalOnBean(Tracer.class)
+@ConditionalOnClass(RestTemplate.class)
+@ConditionalOnProperty(prefix = WebClientTracingProperties.CONFIGURATION_PREFIX, name = "enabled", matchIfMissing = true)
+@AutoConfigureAfter(TracerAutoConfiguration.class)
+@EnableConfigurationProperties(WebClientTracingProperties.class)
+public class RestTemplateTracingAutoConfiguration {
+
+    private static final Log log = LogFactory.getLog(RestTemplateTracingAutoConfiguration.class);
+
+    /**
+     * Provides bean {@link RestTemplateSpanDecorator.StandardTags}.
+     */
+    @Configuration
+    @ConditionalOnMissingBean(RestTemplateSpanDecorator.class)
+    public static class StandardTagsConfiguration {
+
+        @Bean
+        public RestTemplateSpanDecorator.StandardTags standardTagsRestTemplateSpanDecorator() {
```

Comment:
You have same logic available (with a bit different syntax) for `ServletFilterSpanDecorator` and `HandlerInterceptorSpanDecorator`. Since `RestTemplateSpanDecorator` is an interface with one standard implementation, it looks like it should be overridable.
And while I don't currently will be overriding this behavior, I have in mind case where I might need it (as a bridge to legacy system).

Reply:
I guess the behavior to server decorators is a bit different. On the server-side user can disable standard tags whereas here standard tags will be always there.

Reply:
`@ConditionalOnMissingBean(RestTemplateSpanDecorator.class)` on Configuration class makes this config fallback whenever another bean is provided. So behavior is same in the and.
Originally there were more code, that's why it is on Config class instead on Bean. But results is same.